src/UserSettingsStore.js

```diff
@@ -35,6 +35,11 @@ module.exports = {
     },
 
     loadThreePids: function() {
+        if (MatrixClientPeg.get().isGuest()) {
+            return q({
+                threepids: []
+            }); // guests can't poke 3pid endpoint
+        }
```

Comment:
I'm not sure if these conditional guest checks should belong in the JS SDK or not. I'm leaving them wherever is convenient for now until I can think it through more clearly. Putting it in the JS SDK is annoying if we decide to open up endpoints later (like we are with receipts) because it would have to make react SDK contort around the check then.